open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataTopicConnector.java

```diff
@@ -58,15 +64,32 @@
             super(1, 1, Long.MAX_VALUE, TimeUnit.MICROSECONDS, new LinkedBlockingQueue<>(1));
         }
 
+
+        @Override
+        public void afterExecute(Runnable r, Throwable t) {
+            super.afterExecute(r, t);
+
+            /* we don't care why the thread ended , we just restart it */
+            /* The thread will log on exit and on restart already, so no need to let anyone know */
+            initializeProducerAndProducerThread();
+            producerExecutor.execute(producerThread);
+        }
+    }
+    
+    private class KafkaConsumerExecutor extends ThreadPoolExecutor {
+        KafkaConsumerExecutor() {
+            super(1, 1, Long.MAX_VALUE, TimeUnit.MICROSECONDS, new LinkedBlockingQueue<>(1));
+        }
+
+
         @Override
         public void afterExecute(Runnable r, Throwable t) {
             super.afterExecute(r, t);
```

Comment:
What happens if the server is shutting down - should it check that the connector is still active before restarting the threads?

Reply:
Fixed

Reply:
Just need to update the comment so it matches the code and then I think it is ok

Reply:
@mandy-chessell I think I addressed you comment.  I'm not exactly sure which comment you were referring to.